README.md
# Wid Design System

[![style: very good analysis][very_good_analysis_badge]][very_good_analysis_link]
[![Powered by Mason](https://img.shields.io/endpoint?url=https%3A%2F%2Ftinyurl.com%2Fmason-badge)](https://github.com/felangel/mason)
[![License: MIT][license_badge]][license_link]
[![Pub](https://img.shields.io/pub/v/wid_design_system)](https://pub.dev/packages/wid_design_system)

A collection of widgets, colors, styles, providers and asorted utils for implementing the Wid Design System.

## Roadmap

- [ ] Tests
- [ ] More widgets
- [ ] Performance improvements

## How to use

Just import the package and you´ll get a `WidAppTheme.light` light theme and a `WidAppTheme.dark` dark theme out of the box. Check out the example for more detail.

```dart
import 'package:flutter/material.dart';
import 'package:wid_design_system/wid_design_system.dart';

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      darkTheme: WidAppTheme.dark,
      theme: WidAppTheme.light,
      home: const MyHomePage(title: 'My App'),
    );
  }
}
```

You can also add this to your `pubspec.yaml` to include the `Quicksand` font provided by this package:

```yaml
flutter:
  uses-material-design: true
  generate: true

  fonts:
    - family: Quicksand
      fonts:
        - asset: packages/wid_design_system/fonts/Quicksand-Bold.ttf
          weight: 700
        - asset: packages/wid_design_system/fonts/Quicksand-SemiBold.ttf
          weight: 600
        - asset: packages/wid_design_system/fonts/Quicksand-Medium.ttf
          weight: 500
        - asset: packages/wid_design_system/fonts/Quicksand-Regular.ttf
          weight: 400
        - asset: packages/wid_design_system/fonts/Quicksand-Light.ttf
          weight: 300
```

## Installation 💻

**❗ In order to start using Wid Design System you must have the [Flutter SDK][flutter_install_link] installed on your machine.**

Add `wid_design_system` to your `pubspec.yaml`:

```yaml
dependencies:
  wid_design_system:
```

Install it:

```sh
flutter packages get
```

---

## Continuous Integration 🤖

Wid Design System comes with a built-in [GitHub Actions workflow][github_actions_link] powered by [Very Good Workflows][very_good_workflows_link] but you can also add your preferred CI/CD solution.

Out of the box, on each pull request and push, the CI `formats`, `lints`, and `tests` the code. This ensures the code remains consistent and behaves correctly as you add functionality or make changes. The project uses [Very Good Analysis][very_good_analysis_link] for a strict set of analysis options used by our team. Code coverage is enforced using the [Very Good Workflows][very_good_coverage_link].

---

## Running Tests 🧪

For first time users, install the [very_good_cli][very_good_cli_link]:

```sh
dart pub global activate very_good_cli
```

To run all unit tests:

```sh
very_good test --coverage
```

To view the generated coverage report you can use [lcov](https://github.com/linux-test-project/lcov).

```sh
# Generate Coverage Report
genhtml coverage/lcov.info -o coverage/

# Open Coverage Report
open coverage/index.html
```

[flutter_install_link]: https://docs.flutter.dev/get-started/install
[github_actions_link]: https://docs.github.com/en/actions/learn-github-actions
[license_badge]: https://img.shields.io/badge/license-MIT-blue.svg
[license_link]: https://opensource.org/licenses/MIT
[logo_black]: https://raw.githubusercontent.com/VGVentures/very_good_brand/main/styles/README/vgv_logo_black.png#gh-light-mode-only
[logo_white]: https://raw.githubusercontent.com/VGVentures/very_good_brand/main/styles/README/vgv_logo_white.png#gh-dark-mode-only
[mason_link]: https://github.com/felangel/mason
[very_good_analysis_badge]: https://img.shields.io/badge/style-very_good_analysis-B22C89.svg
[very_good_analysis_link]: https://pub.dev/packages/very_good_analysis
[very_good_cli_link]: https://pub.dev/packages/very_good_cli
[very_good_coverage_link]: https://github.com/marketplace/actions/very-good-coverage
[very_good_ventures_link]: https://verygood.ventures
[very_good_ventures_link_light]: https://verygood.ventures#gh-light-mode-only
[very_good_ventures_link_dark]: https://verygood.ventures#gh-dark-mode-only
[very_good_workflows_link]: https://github.com/VeryGoodOpenSource/very_good_workflows


CHANGELOG.md
# 1.0.0
- MASSIVE BREAKING CHANGES
- breaking-change: Most theme clases changed from having static members to have instance members.
- feat: Added new WidAppTheme factory constructor "High contrast black and white".

# 0.2.7
- feat: Implemented DatePickerTheme 🎉
- feat: Implemented TimePickerTheme 🎉
- feat: Improved color coexistence 🎉

# 0.2.6
- hot-fix: Fixed button theme colors

# 0.2.5
- feat: Improved color coexistence 🎉

# 0.2.4
- feat: Implemented ChipTheme 🎉
- feat: Implemented BottomSheetTheme 🎉

# 0.2.3
- hot-fix: Misspelling in Appbar theme
- feat: Replaced grey1-4 with n100-900 palette color
- feat: Implemented MaterialColor color swatch with pink 🎉

# 0.2.2
- hot-fix: Minor adjustment in input text´s label styles

# 0.2.1
- feat: Added WidTapToHideKeyboard widget 🎉
- feat: Implemented AppBarThemes, BottomAppBarThemes, CheckboxThemes, PopupMenuThemes🎉
- feat: Added widgets to example view 🎉
# 0.2.0

- feat: Added custom button with 4 variants 🎉
- feat: Added several theme implementations for several widgets 🎉
- feat: Added custom touchable widget 🎉
- feat: Exposing and providing fonts 🎉

# 0.1.0-beta.1

- feat: initial commit 🎉


lib/src/theme/palette.dart
import 'package:flutter/material.dart';

/// Primary color
const Color _primary = Color(0xFFFF82B8);

final Map<int, Color> _pink = {
  50: const Color(0xFFFDCDE2),
  100: const Color(0xFFFEBDDA),
  200: const Color(0xFFFEA6CC),
  300: const Color(0xFFFF8EBF),
  400: const Color(0xFFFC94C1),
  500: _primary,
  600: const Color(0xFFFD71AD),
  700: const Color(0xFFFF509C),
  800: const Color(0xFFFD2B86),
  900: const Color(0xFFFF006F),
};

/// Color palette for Wid Design System
class WidAppColors {
  /// Call-to-action color
  static const Color dark = Color(0xFF1C1C1C);

  /// Call-to-action color
  static const Color light = Color(0xFFFEFEFE);

  /// Call-to-action color
  static const Color callToAction = Color(0xFFFF82B8);

  /// Color for success/completion feedback
  static const Color success = Color(0xFF189215);

  /// Color for info/warning feedback
  static const Color info = Color(0xFFFFC914);

  /// Color for error/alert feedback
  static const Color error = Color(0xFFCC0B15);

  /// Custom yellow color
  static const Color yellow = Color(0xFFFFC914);

  /// Custom white color
  static const Color white = Color(0xFFFEFEFE);

  /// Custom black color
  static const Color black = Color(0xFF1C1C1C);

  /// Custom grey color (shade 100)
  static const Color n100 = Color(0xFFF6F6F6);

  /// Custom grey color (shade 200)
  static const Color n200 = Color(0xFFEAEAEA);

  /// Custom grey color (shade 300)
  static const Color n300 = Color(0xFFD2D3D3);

  /// Custom grey color (shade 400)
  static const Color n400 = Color(0xFFB1B2B2);

  /// Custom grey color (shade 500)
  static const Color n500 = Color(0xFF8E9090);

  /// Custom grey color (shade 600)
  static const Color n600 = Color(0xFF7A7A7A);

  /// Custom grey color (shade 700)
  static const Color n700 = Color(0xFF6B6B6B);

  /// Custom grey color (shade 800)
  static const Color n800 = Color(0xFF363939);

  /// Custom grey color (shade 900)
  static const Color n900 = Color(0xFF252729);

  /// Background color
  static const Color background = white;

  /// Primary color
  static MaterialColor primary = MaterialColor(_primary.value, _pink);
}

/// Material color based on Black for black and white theme
MaterialColor materialColorBlack = const MaterialColor(
  0xFF000000,
  <int, Color>{
    50: Color(0xFF000000),
    100: Color(0xFF000000),
    200: Color(0xFF000000),
    300: Color(0xFF000000),
    400: Color(0xFF000000),
    500: Color(0xFF000000),
    600: Color(0xFF000000),
    700: Color(0xFF000000),
    800: Color(0xFF000000),
    900: Color(0xFF000000),
  },
);

/// Material color based on White for black and white theme
MaterialColor materialColorWhite = const MaterialColor(
  0xFFFFFFFF,
  <int, Color>{
    50: Color(0xFFFFFFFF),
    100: Color(0xFFFFFFFF),
    200: Color(0xFFFFFFFF),
    300: Color(0xFFFFFFFF),
    400: Color(0xFFFFFFFF),
    500: Color(0xFFFFFFFF),
    600: Color(0xFFFFFFFF),
    700: Color(0xFFFFFFFF),
    800: Color(0xFFFFFFFF),
    900: Color(0xFFFFFFFF),
  },
);
